scrapy/addons.py

```diff
@@ -41,13 +29,29 @@ def load_settings(self, settings: Settings) -> None:
             which to read the add-on configuration
         :type settings: :class:`~scrapy.settings.Settings`
         """
-        addons = build_component_list(settings["ADDONS"])
-        for addon in build_component_list(settings["ADDONS"]):
-            self._add(addon, settings)
+        enabled = []
+        for clspath in build_component_list(settings["ADDONS"]):
+            try:
+                addoncls = load_object(clspath)
+                addon = create_instance(
+                    addoncls, settings=settings, crawler=self.crawler
+                )
+                addon.update_settings(settings)
+                self.addons.append(addon)
+            except NotConfigured as e:
+                if e.args:
+                    clsname = (
+                        clspath.split(".")[-1] if isinstance(clspath, str) else clspath
```

Comment:
Why is this processing needed?

Reply:
No strong opinion. 1 sounds great, if it is not too bothersome. Otherwise, 2 makes sense. 3 is more consistent than 2, but I don’t think that matters that much here.

Reply:
It looks like you went with 1 :)

Reply:
Yeah, I decided it's easy enough.